pkg/deploy/reaper/reaper.go

```diff
@@ -84,10 +106,34 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, timeout time.
 			err := reaper.kc.Pods(pod.Namespace).Delete(pod.Name, gracePeriod)
 			if err != nil {
 				// Better not error out here...
-				glog.Infof("Cannot delete Pod %s/%s: %v", pod.Namespace, pod.Name, err)
+				glog.Infof("Cannot delete lifecycle Pod %s/%s for deployment %s/%s: %v", pod.Namespace, pod.Name, namespace, name, err)
 			}
 		}
 	}
 
-	return nil
+	return reaper.oc.DeploymentConfigs(namespace).Delete(name)
+}
+
+type updateConfigFunc func(d *deployapi.DeploymentConfig)
+
+func (reaper *DeploymentConfigReaper) updateDeploymentWithRetries(namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
+	var (
+		config *deployapi.DeploymentConfig
+		err    error
+	)
+	deploymentConfigs := reaper.oc.DeploymentConfigs(namespace)
+	resultErr := wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
+		config, err = deploymentConfigs.Get(name)
+		if err != nil {
+			return false, err
+		}
+		// Apply the update, then attempt to push it to the apiserver.
+		applyUpdate(config)
+		config, err = deploymentConfigs.Update(config)
+		if err != nil {
```

Comment:
@kargakis this is from upstream, if we want that I will fix this in upstream as well.